Tools/cases_generator/generators_common.py

```diff
@@ -177,6 +177,35 @@ def replace_check_eval_breaker(
     if not uop.properties.ends_with_eval_breaker:
         out.emit_at("CHECK_EVAL_BREAKER();", tkn)
 
+def replace_pystackref_frompyobjectnew(
+    out: CWriter,
+    tkn: Token,
+    tkn_iter: Iterator[Token],
+    uop: Uop,
+    stack: Stack,
+    inst: Instruction | None,
+) -> None:
+    out.emit(tkn)
+    emit_to(out, tkn_iter, "SEMI")
+    out.emit(";\n")
+    target = uop.body.index(tkn)
+    assert uop.body[target-1].kind == "EQUALS", (f"{uop.name} Result of a specials that is flushed"
+                                                        " must be written to a stack variable directly")
+    # Scan to look for the first thing it's assigned to
+    found_valid_assignment = ""
+    for assgn_target in reversed(uop.body[:target-1]):
+        out_names = [out.name for out in uop.stack.outputs]
+        if assgn_target.kind == "IDENTIFIER":
+            if assgn_target.text in out_names:
+                found_valid_assignment = assgn_target.text
+                break
+
+    if found_valid_assignment:
```

Comment:
What happens if we haven't found a valid assignment?
That sounds like an error.

Reply:
This now raises an `analysis_error`.